overlays/holo-nixpkgs/hpos-led-manager/src/main.rs

```diff
@@ -53,17 +86,53 @@ fn main() -> Fallible<()> {
 
         let hpos_config_found = Path::new("/run/hpos-init/hpos-config.json").exists();
 
-        let state = match (online, hpos_config_found) {
-            (false, _) => State::Flash(Color::Purple),
-            (true, false) => State::Static(Color::Blue),
+        if counter % HYDRA_POLLING_INTERVAL == 0 {  
+            revision_json = get_hydra_revision()?;
+            hydra_revision = revision_json["jobsetevalinputs"]["holo-nixpkgs"]["revision"]
+                .as_str()
+                .context("revision is not a string")?;
+            counter = 0;
+        }
+        counter += 1;
+
+        let local_revision = fs::read_to_string("/root/.nix-revision")
+            .expect("Something went wrong reading nix-revision");
+        let update_required = local_revision != hydra_revision; // If this lights up then it's likely the updater isn't working properly
+
+        let mut tls_certificate_valid = false;
+        let tls_certificate_found = Path::new("/var/lib/acme/default/account_reg.json").exists();
+        if tls_certificate_found {
+            let tls_certificate_raw = fs::read_to_string("/var/lib/acme/default/account_reg.json")
+            .expect("Something went wrong reading certificate");
+            let tls_certificate: Value = serde_json::from_str(&tls_certificate_raw)?;
+            tls_certificate_valid = tls_certificate["body"]["status"] == "valid"; // Note that the TLS_certificate returns a borrow (&value)
+        } 
```

Comment:
Avoid using `mut`able variables if you can.
Remember that `if` is also an expression, and has a value.

```suggestion
        let tls_certificate_found = Path::new("/var/lib/acme/default/account_reg.json").exists();
        let tls_certificate_valid = if tls_certificate_found {
            let tls_certificate_raw = fs::read_to_string("/var/lib/acme/default/account_reg.json")
            .expect("Something went wrong reading certificate");
            let tls_certificate: Value = serde_json::from_str(&tls_certificate_raw)?;
            tls_certificate["body"]["status"] == "valid" // Note that the TLS_certificate returns a borrow (&value)
        } else {
        	false
        }; 
```